lib/puppet/functions/foreman_proxy/get_network_in_addr.rb

```diff
@@ -1,21 +1,22 @@
-module Puppet::Parser::Functions
-  newfunction(:get_network_in_addr, :type => :rvalue,
-              :doc => 'Get the in-addr.arpa notation for the given IP address and a netmask'
-  ) do |args|
-    require 'ipaddr'
-
-    if args.length != 2 then
-      raise Puppet::ParseError, ("get_network_in_addr(): must give an address and a netmask)")
-    end
+# Get the in-addr.arpa notation for the given IP address and a netmask
+
+require 'ipaddr'
+
+Puppet::Functions.create_function(:'foreman_proxy::get_network_in_addr') do
+  dispatch :get_network_in_addr do
+    required_param 'String', :addr
```

Comment:
I'd prefer full names (```address``` and ```netmask```) for readability. Also wondering if we should use the ip address type from stdlib and remove the begin/rescue code around the ```IPAddr.new``` calls.